### PR TITLE
[incubator-kie-issues#1209] Enable scesim for DRL in kogito

### LINKED ISSUE #1209: Enable scesim for DRL in kogito
Enable scesim execution for rules inside kogito project.
EVentually implement a custo KogitoDRLRunner - see equivalent DMN one.

@yesamer @baldimir

### PR DESCRIPTION
**Pull request by gitgabrio**
Fixes https://github.com/apache/incubator-kie-issues/issues/1209

Depends on https://github.com/apache/incubator-kie-drools/pull/5954

This PR creates quarkus and springboot rules+scesim kogito examples.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

- for <b>pull request checks</b>  
  Please add comment: <b>Jenkins retest this</b>

- for a <b>specific pull request check</b>  
  Please add comment: <b>Jenkins (re)run [kogito-examples] tests</b>

- for <b>quarkus branch checks</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins run quarkus-branch</b>

- for a <b>quarkus branch specific check</b>  
  Run checks against Quarkus current used branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-branch</b>

- for <b>quarkus main checks</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins run quarkus-main</b>

- for a <b>specific quarkus main check</b>  
  Run checks against Quarkus main branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-main</b>

- for <b>quarkus lts checks</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins run quarkus-lts</b>

- for a <b>specific quarkus lts check</b>  
  Run checks against Quarkus lts branch  
  Please add comment: <b>Jenkins (re)run [kogito-examples] quarkus-lts</b>
 
- for <b>native checks</b>  
  Run native checks  
  Please add comment: <b>Jenkins run native</b>

- for a <b>specific native check</b>  
  Run native checks 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native</b>

- for <b>native lts checks</b>  
  Run native checks against quarkus lts branch
  Please add comment: <b>Jenkins run native-lts</b>

- for a <b>specific native lts check</b>  
  Run native checks against quarkus lts branch 
  Please add comment: <b>Jenkins (re)run [kogito-examples] native-lts</b>
 
</details>

<details>
<summary>
How to backport a pull request to a different branch?
</summary>

In order to automatically create a **backporting pull request** please add one or more labels having the following format `backport-<branch-name>`, where `<branch-name>` is the name of the branch where the pull request must be backported to (e.g., `backport-7.67.x` to backport the original PR to the `7.67.x` branch).

> **NOTE**: **backporting** is an action aiming to move a change (usually a commit) from a branch (usually the main one) to another one, which is generally referring to a still maintained release branch. Keeping it simple: it is about to move a specific change or a set of them from one branch to another.

Once the original pull request is successfully merged, the automated action will create one backporting pull request per each label (with the previous format) that has been added.

If something goes wrong, the author will be notified and at this point a manual backporting is needed.

> **NOTE**: this automated backporting is triggered whenever a pull request on `main` branch is labeled or closed, but both conditions must be satisfied to get the new PR created.
</details>

### CODE REVIEW

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/java/org/kie/kogito/legacy/Hello.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+package org.kie.kogito.legacy;
```
Comment (porcelli) [inline kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/java/org/kie/kogito/legacy/Hello.java]: Missing Apache header

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/README.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
+# Rules with Legacy API + Quarkus example
```
Comment (porcelli) [inline kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/README.md:19]: Missing Apache header

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/java/org/kie/kogito/legacy/Message.java [hunk 3]
```diff
@@ -0,0 +1,14 @@
+package org.kie.kogito.legacy;
```
Comment (porcelli) [inline kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/java/org/kie/kogito/legacy/Message.java]: missing Apache header

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/resources/org/kie/kogito/legacy/HelloRule.drl [hunk 4]
```diff
@@ -0,0 +1,17 @@
+package org.kie.kogito.legacy;
```
Comment (porcelli) [inline kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/resources/org/kie/kogito/legacy/HelloRule.drl:19]: Missing Apache header

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/java/testscenario/ScenarioJunitActivatorTest.java [hunk 5]
```diff
@@ -0,0 +1,11 @@
+package testscenario;
```
Comment (porcelli) [inline kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/java/testscenario/ScenarioJunitActivatorTest.java:19]: Missing Apache header

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/resources/org/kie/kogito/legacy/HelloScesim.scesim [hunk 6]
```diff
@@ -0,0 +1,273 @@
+<ScenarioSimulationModel version="1.8">
```
Comment (porcelli) [inline kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/resources/org/kie/kogito/legacy/HelloScesim.scesim:20]: is Apache header supported?

File: kogito-springboot-examples/rules-legacy-scesim-springboot-example/README.md [hunk 7]
```diff
@@ -0,0 +1,89 @@
+# RuleUnit + Spring Boot example
```
Comment (porcelli) [inline kogito-springboot-examples/rules-legacy-scesim-springboot-example/README.md:19]: Missing Apache header

File: kogito-springboot-examples/rules-legacy-scesim-springboot-example/src/main/java/org/kie/kogito/legacy/Hello.java [hunk 8]
```diff
@@ -0,0 +1,41 @@
+package org.kie.kogito.legacy;
```
Comment (porcelli) [inline kogito-springboot-examples/rules-legacy-scesim-springboot-example/src/main/java/org/kie/kogito/legacy/Hello.java]: Missing Apache header

**Review by porcelli — CHANGES_REQUESTED** [review]

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/resources/org/kie/kogito/legacy/HelloScesim.scesim (hunk 6 above)
Comment (yesamer) [reply kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/resources/org/kie/kogito/legacy/HelloScesim.scesim:20]: @gitgabrio Can you please add the prolog as well? `<?xml version="1.0" encoding="UTF-8"?>`

**Review by yesamer — COMMENTED** [review]
@gitgabrio Excellent work! What do you think about adding one example for the new API? (Kogito) Maybe we can use an existing rule example for that (eg. `dmn-quarkus-example` contains a scesim file)

**Review by mariofusco — APPROVED** [review]

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/resources/META-INF/kmodule.xml [hunk 9]
```diff
@@ -0,0 +1 @@
+<kmodule xmlns="http://www.drools.org/xsd/kmodule"/>
```
Comment (porcelli) [inline kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/main/resources/META-INF/kmodule.xml:19]: Headers?

**Comment by gitgabrio** [comment]
@porcelli @yesamer 
I add the missing headers, but:
1. what should be the format for Readme.md files ?
2. no other Readme.md has the header
3. should also kmodule.xml and scesim have the header ? No other have it

**Comment by kie-ci3** [comment]
**PR job** `#1` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-examples -u https://github.com/apache/incubator-kie-kogito-examples/pull/1921 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1921/1/display/redirect

**Test results:**
- PASSED: 170
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1921/1/testReport/org.kie.kogito.examples.quarkus/GrafanaDockerComposeIT/testGrafanaDashboards/">org.kie.kogito.examples.quarkus.GrafanaDockerComposeIT.testGrafanaDashboards</a></summary>
1 expectation failed.<br/>JSON path title doesn't match.<br/>Expected: a collection containing "dmn-drools-quarkus-metrics_999-SNAPSHOT - hello - Operational Dashboard"<br/>  Actual: <[dmn-drools-quarkus-metrics_999-SNAPSHOT - Custom - Domain Dashboard, dmn-drools-quarkus-metrics_999-SNAPSHOT - Custom - Operational Dashboard, dmn-drools-quarkus-metrics_999-SNAPSHOT - Hello - Domain Dashboard, dmn-drools-quarkus-metrics_999-SNAPSHOT - LoanEligibility - Domain Dashboard]><br/>
</details>

File: kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/resources/org/kie/kogito/legacy/HelloScesim.scesim (hunk 6 above)
Comment (porcelli) [reply kogito-quarkus-examples/rules-legacy-scesim-quarkus-example/src/test/resources/org/kie/kogito/legacy/HelloScesim.scesim:20]: headers still missing

File: kogito-springboot-examples/rules-legacy-scesim-springboot-example/src/main/resources/META-INF/kmodule.xml [hunk 10]
```diff
@@ -0,0 +1 @@
+<kmodule xmlns="http://www.drools.org/xsd/kmodule" />
```
Comment (porcelli) [inline kogito-springboot-examples/rules-legacy-scesim-springboot-example/src/main/resources/META-INF/kmodule.xml:19]: header?

**Comment by gitgabrio** [comment]
> @gitgabrio Excellent work! What do you think about adding one example for the new API? (Kogito) Maybe we can use an existing rule example for that (eg. `dmn-quarkus-example` contains a scesim file)

I would prefer to postpone after  https://github.com/apache/incubator-kie/issues/7071 (and possibly https://github.com/apache/incubator-kie/issues/7070)

File: kogito-springboot-examples/rules-legacy-scesim-springboot-example/src/test/resources/HelloScesim.scesim [hunk 11]
```diff
@@ -0,0 +1,274 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<ScenarioSimulationModel version="1.8">
```
Comment (porcelli) [inline kogito-springboot-examples/rules-legacy-scesim-springboot-example/src/test/resources/HelloScesim.scesim:20]: Header?

**Review by porcelli — CHANGES_REQUESTED** [review]

**Comment by porcelli** [comment]
examples repo has been behind on effort to comply with Apache as it's not part of the official release.

But as new code lands... we need to make sure it comply with all Apache rules

**Comment by gitgabrio** [comment]
> examples repo has been behind on effort to comply with Apache as it's not part of the official release.
> 
> But as new code lands... we need to make sure it comply with all Apache rules

https://github.com/apache/incubator-kie-kogito-examples/pull/1921#issuecomment-2115131781

**Comment by porcelli** [comment]
everything that doesn't break should have a header

**Review by porcelli — APPROVED** [review]

**Review by yesamer — APPROVED** [review]

**Comment by gitgabrio** [comment]
Jenkins retest this

**Comment by kie-ci3** [comment]
**PR job** `#4` was: **UNSTABLE**
Possible explanation: This should be test failures


<details>
<summary><b>Reproducer</b></summary>



build-chain build full_downstream  -f 'https://raw.githubusercontent.com/${AUTHOR:apache}/incubator-kie-kogito-pipelines/${BRANCH:main}/.ci/buildchain-config-pr-cdb.yaml' -o 'bc' -p apache/incubator-kie-kogito-examples -u https://github.com/apache/incubator-kie-kogito-examples/pull/1921 --skipParallelCheckout

NOTE: To install the build-chain tool, please refer to https://github.com/kiegroup/github-action-build-chain#local-execution

</details>


Please look here: https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1921/4/display/redirect

**Test results:**
- PASSED: 391
- FAILED: 1

Those are the test failures: 
<details>
<summary><a href="https://ci-builds.apache.org/job/KIE/job/kogito/job/main/job/pullrequest_jobs/job/kogito-examples-pr/job/PR-1921/4/testReport/org.acme/AcmeExchangeResourceIT/unauthorizedTest/">org.acme.AcmeExchangeResourceIT.unauthorizedTest</a></summary>
java.lang.RuntimeException: io.quarkus.builder.BuildException: Build failure: Build failed due to errors<br/> [error]: Build step io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor#startKeycloakContainer threw an exception: java.lang.RuntimeException: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:23.0.7<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:249)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)<br/> at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)<br/> at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)<br/> at java.base/java.lang.reflect.Method.invoke(Method.java:568)<br/> at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)<br/> at io.quarkus.builder.BuildContext.run(BuildContext.java:256)<br/> at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)<br/> at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)<br/> at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)<br/> at java.base/java.lang.Thread.run(Thread.java:833)<br/> at org.jboss.threads.JBossThread.run(JBossThread.java:501)<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Container startup failed for image quay.io/keycloak/keycloak:23.0.7<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:359)<br/> at org.testcontainers.containers.GenericContainer.start(GenericContainer.java:330)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.lambda$startContainer$4(KeycloakDevServicesProcessor.java:376)<br/> at java.base/java.util.Optional.orElseGet(Optional.java:364)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startContainer(KeycloakDevServicesProcessor.java:401)<br/> at io.quarkus.oidc.deployment.devservices.keycloak.KeycloakDevServicesProcessor.startKeycloakContainer(KeycloakDevServicesProcessor.java:197)<br/> ... 11 more<br/>Caused by: org.rnorth.ducttape.RetryCountExceededException: Retry limit hit with exception<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:88)<br/> at org.testcontainers.containers.GenericContainer.doStart(GenericContainer.java:344)<br/> ... 16 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Could not create/start container<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:563)<br/> at org.testcontainers.containers.GenericContainer.lambda$doStart$0(GenericContainer.java:354)<br/> at org.rnorth.ducttape.unreliables.Unreliables.retryUntilSuccess(Unreliables.java:81)<br/> ... 17 more<br/>Caused by: org.testcontainers.containers.ContainerLaunchException: Timed out waiting for log output matching '.*Keycloak.*started.*'<br/> at org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy.waitUntilReady(LogMessageWaitStrategy.java:47)<br/> at org.testcontainers.containers.wait.strategy.AbstractWaitStrategy.waitUntilReady(AbstractWaitStrategy.java:52)<br/> at org.testcontainers.containers.GenericContainer.waitUntilContainerStarted(GenericContainer.java:909)<br/> at org.testcontainers.containers.GenericContainer.tryStart(GenericContainer.java:500)<br/> ... 19 more<br/>
</details>

**Comment by porcelli** [comment]
failed test isn't related to the changes

**Pull request merged**